hpc_provisioner/src/hpc_provisioner/handlers.py

```diff
@@ -57,11 +65,33 @@ def pcluster_handler(event, _context=None):
 def pcluster_create_request_handler(event, _context=None):
     """Request the creation of an HPC cluster for a given vlab_id and project_id"""
 
-    vlab_id, project_id, _ = _get_vlab_query_params(event)
+    vlab_id, project_id, _, _ = _get_vlab_query_params(event)
+    ec2_client = boto3.client("ec2")
+    sm_client = boto3.client("secretsmanager")
+    ssh_keypair = create_keypair(
+        ec2_client,
+        vlab_id=vlab_id,
+        project_id=project_id,
+        tags=[
+            {"Key": VLAB_TAG_KEY, "Value": vlab_id},
+            {"Key": PROJECT_TAG_KEY, "Value": project_id},
+            {"Key": BILLING_TAG_KEY, "Value": BILLING_TAG_VALUE},
+        ],
+    )
+
+    if "KeyMaterial" in ssh_keypair:
+        secret = create_secret(
+            sm_client, vlab_id, project_id, ssh_keypair["KeyName"], ssh_keypair["KeyMaterial"]
+        )
+    else:
+        secret = get_secret(sm_client, ssh_keypair["KeyName"])
```

Comment:
Seems like this could be done in `create_keypair`, too?

Reply:
It feels like that would make `create_keypair` do a bit too much. I could move the get-or-create logic to `create_secret` and rename it to `create_or_get_secret` to get it out of `pcluster_create_request_handler` though.